docs/reference/icalendar.rst

```diff
@@ -0,0 +1,35 @@
+icalendar package
+=================
+
+.. automodule:: icalendar
+   :members:
+   :show-inheritance:
+   :undoc-members:
+
+Subpackages
+-----------
+
+.. toctree::
+   :maxdepth: 4
+
+   icalendar.timezone
```

Comment:
I personally would not distinct between modules and sub packages as they should not be treated differently. As a rule for icalendar it seems: Everything that is offered publicly can be imported with `from icalendar import ...`.
The sub-packages are just to have smaller files.
In the future, I might split up the `prop.py` and the `cal.py` file. I wonder if you would like to move them in the docs then.

What benefit do you see in splitting package/module?

Reply:
This is the default output from running `sphinx-apidoc -o reference/ ../src/` in an activated virtual environment. There are options that I can issue according to https://www.sphinx-doc.org/en/master/usage/extensions/apidoc.html. If we can find the options that do what you want automatically, then we should use those. Otherwise, we'll need to do some manual adjustment of the API .rst files.

Reply:
I should also say that there are other options for documenting the API, including https://sphinx-autoapi.readthedocs.io/en/latest/, but I don't know if it does what you want any better than sphinx-apidoc.

I don't do much development with icalendar. I honestly don't know or have an opinion about what is the best way to present the API docs, and I'm looking for guidance.

Reply:
I would say, any way you decide is alright. In the end, people will look at the API documentation and that has code examples and that is what they copy anyway. So, the structure seems alright to me now, looking at it. Uncoupling reference structure and module structure in my mind. They serve different purposes.

Reply:
In the latest version of the API docs, I used the following command:

```shell
../.venv/bin/sphinx-apidoc -f -M -e -a --remove-old -o reference ../src ../src/icalendar/tests
```

`../src/icalendar/tests` is excluded. @niccokunzmann do you want to include them? I don't know if they're useful as an API reference, and a contributor would look for examples in code anyway, which is why I excluded them.

For additional options, see https://www.sphinx-doc.org/en/master/man/sphinx-apidoc.html#options.